projects/auth0-angular/src/lib/auth.config.ts

```diff
@@ -52,6 +64,15 @@ export interface HttpInterceptorRouteConfig {
    * access token to attach to the outgoing request.
    */
   tokenOptions?: GetTokenSilentlyOptions;
+
+  /**
+   * The HTTP method to match on. If specified, the HTTP method of
+   * the outgoing request will be checked against this. If there is no match, the
+   * Authorization header is not attached.
+   *
+   * The HTTP method name is case-sensitive.
+   */
+  httpMethod?: HttpMethod | string;
```

Comment:
do we want to keep this `string` out of the API for now, until someone requires to use a method we don't have in the enum? Thinking removing it later will be a breaking change, and that supported methods should be those that the angular HTTP client supports.

```suggestion
  httpMethod?: HttpMethod;
```

Reply:
I don't have a particularly strong opinion on this, but I will say that Angular does let you pass _any_ verb as string [using `request`](https://www.typescriptlang.org/docs/handbook/enums.html). If we removed support for string, we'd already be limiting customers needlessly if they need that option.

Reply:
That's a fair point.